modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/SAMLIdPInitiatedSLOTestCase.java

```diff
@@ -0,0 +1,609 @@
+                    .get(APPLICATION_ONE).getArtifact(), user.getUsername(), user.getPassword(), httpClient);
+
+            if (requestMissingClaims(response)) {
+                String pastrCookie = Utils.getPastreCookie(response);
+                Assert.assertNotNull(pastrCookie, "pastr cookie not found in response.");
+                EntityUtils.consume(response.getEntity());
+
+                response = Utils.sendPOSTConsentMessage(response, COMMON_AUTH_URL, USER_AGENT, String.format(ACS_URL
+                        , applications.get(APPLICATION_ONE).getArtifact()), httpClient, pastrCookie);
+                EntityUtils.consume(response.getEntity());
+            }
+
+            String redirectUrl = Utils.getRedirectUrl(response);
+            if (StringUtils.isNotBlank(redirectUrl)) {
+                response = Utils.sendRedirectRequest(response, USER_AGENT, ACS_URL, applications.get(APPLICATION_ONE)
+                        .getArtifact(), httpClient);
+            }
+
+            String samlResponse = Utils.extractDataFromResponse(response, CommonConstants.SAML_RESPONSE_PARAM, 5);
+            EntityUtils.consume(response.getEntity());
+
+            response = sendSAMLMessage(String.format(ACS_URL, applications.get(APPLICATION_ONE).getArtifact()),
+                    CommonConstants.SAML_RESPONSE_PARAM, samlResponse);
+            resultPage = DataExtractUtil.getContentData(response);
+
+            Assert.assertTrue(resultPage.contains("You are logged in as " + user.getTenantAwareUsername()),
+                    "SAML SSO Login failed for " + applications.get(APPLICATION_ONE).getArtifact());
+
+        } catch (Exception e) {
+            Assert.fail("SAML SSO Login test failed for " + applications.get(APPLICATION_ONE).getArtifact(), e);
+        }
+
+        doSSOtoAppTwo();
+    }
+
+    @Test(alwaysRun = true, description = "Testing SAML IdP initiated SLO", groups = "wso2.is",
+            dependsOnMethods = {"testSAMLSSOLogin"})
+    public void testSAMLIdpInitiatedSLO() throws Exception {
+
+        try {
+            logAdmin.updateLoggerData("org.wso2.carbon.identity.sso.saml.logout.LogoutRequestSender",
+                    LoggingAdminClient.LogLevel.DEBUG.name(), true, false);
+            logViewer.clearLogs();
+
+            HttpResponse response = Utils.sendGetRequest(SAML_IDP_SLO_URL, USER_AGENT, httpClient);
+            String resultPage = DataExtractUtil.getContentData(response);
+
+            Assert.assertTrue(resultPage.contains("You have successfully logged out") &&
+                    !resultPage.contains("error"), "SAML IdP initiated SLO failed for " +
+                    applications.get(APPLICATION_ONE).getArtifact() + " & " +
+                    applications.get(APPLICATION_TWO).getArtifact());
+
+            boolean requestOneSentLogFound = checkForLog(logViewer,
+                    "single logout request is sent to : http://localhost:8490/travelocity.com/home.jsp is " +
```

Comment:
There are few places where we use the same URL . Can you get it to a constant and reuse ?